File: app/ui/src/main/java/com/fsck/k9/activity/FolderInfoHolder.java
package com.fsck.k9.activity;


import com.fsck.k9.Account;
import com.fsck.k9.mailstore.Folder;
import com.fsck.k9.mailstore.FolderType;
import com.fsck.k9.mailstore.LocalFolder;
import com.fsck.k9.ui.folders.FolderNameFormatter;


public class FolderInfoHolder {
    private final FolderNameFormatter folderNameFormatter;

    public final String serverId;
    public final String displayName;
    public final long lastChecked;
    public boolean loading;
    public boolean moreMessages;


    public FolderInfoHolder(FolderNameFormatter folderNameFormatter, LocalFolder localFolder, Account account) {
        this.folderNameFormatter = folderNameFormatter;
        this.serverId = localFolder.getServerId();
        this.lastChecked = localFolder.getLastUpdate();
        this.displayName = getDisplayName(account, localFolder);
        moreMessages = localFolder.hasMoreMessages();
    }

    private String getDisplayName(Account account, LocalFolder localFolder) {
        String serverId = localFolder.getServerId();
        Folder folder = new Folder(
                localFolder.getDatabaseId(),
                serverId,
                localFolder.getName(),
                getFolderType(account, serverId));

        return folderNameFormatter.displayName(folder);
    }

    public static FolderType getFolderType(Account account, String serverId) {
        if (serverId.equals(account.getInboxFolder())) {
            return FolderType.INBOX;
        } else if (serverId.equals(account.getOutboxFolder())) {
            return FolderType.OUTBOX;
        } else if (serverId.equals(account.getArchiveFolder())) {
            return FolderType.ARCHIVE;
        } else if (serverId.equals(account.getDraftsFolder())) {
            return FolderType.DRAFTS;
        } else if (serverId.equals(account.getSentFolder())) {
            return FolderType.SENT;
        } else if (serverId.equals(account.getSpamFolder())) {
            return FolderType.SPAM;
        } else if (serverId.equals(account.getTrashFolder())) {
            return FolderType.TRASH;
        } else {
            return FolderType.REGULAR;
        }
    }
}

File: app/ui/src/main/java/com/fsck/k9/activity/FolderInfoHolder.kt
package com.fsck.k9.activity

import com.fsck.k9.Account
import com.fsck.k9.mailstore.Folder
import com.fsck.k9.mailstore.FolderType
import com.fsck.k9.mailstore.LocalFolder
import com.fsck.k9.ui.folders.FolderNameFormatter

class FolderInfoHolder(
    private val folderNameFormatter: FolderNameFormatter,
    localFolder: LocalFolder,
    account: Account
) {
    @JvmField val databaseId = localFolder.databaseId
    @JvmField val serverId: String = localFolder.serverId
    @JvmField val displayName = getDisplayName(account, localFolder)
    @JvmField var loading = false
    @JvmField var moreMessages = localFolder.hasMoreMessages()

    private fun getDisplayName(account: Account, localFolder: LocalFolder): String {
        val serverId = localFolder.serverId
        val folder = Folder(
            localFolder.databaseId,
            serverId,
            localFolder.name,
            getFolderType(account, serverId)
        )
        return folderNameFormatter.displayName(folder)
    }

    companion object {
        @JvmStatic
        fun getFolderType(account: Account, serverId: String): FolderType {
            return when (serverId) {
                account.inboxFolder -> FolderType.INBOX
                account.outboxFolder -> FolderType.OUTBOX
                account.archiveFolder -> FolderType.ARCHIVE
                account.draftsFolder -> FolderType.DRAFTS
                account.sentFolder -> FolderType.SENT
                account.spamFolder -> FolderType.SPAM
                account.trashFolder -> FolderType.TRASH
                else -> FolderType.REGULAR
            }
        }
    }
}
